### ISSUE TITLE
MAX Request - Decal projection

### ISSUE THREAD

**Issue by fearlesswee** [TheGameCreators/GameGuruRepo#711]
**This is a request for a new feature in MAX.**

The ability to project decals onto surfaces in GG Max would be a tremendously useful addition in many areas. Some examples of how decal projection could be used:

- To add blood splatter/spray onto surfaces when a character is damaged

- Bullet holes on surfaces when shot

- Additional detail for level design without having to make a new texture variation of any model(s)

- Etc.

[Here is a link to an image to illustrate what decal projection is](http://i.imgur.com/zdLLfvf.png)

It would be important to allow decals to be both projected in-editor _as well as via script_

**Comment by ErrantAI** [comment]
Yes!  Please!
It's a shame that even bullet hole/splatter/scorch decals were dropped for GG.  Would love to see this make a return and being able to detail the environment via projected textures would be an amazing addition.

**Comment by michaeloldman** [comment]
Yeah, AmenMoses had bullet holes working using particles, could add that to MAX.

**Comment by AmenMoses** [comment]
Although it can be done with particles there are issues with animated entities as you really need the image to move with the animation which can't really be done in Lua.
I imagine this would also need a special shader to support projecting a flat image onto the surface of entities and the terrain properly.

**Comment by rvanner** [comment]
@fearlesswee Thanks for your ideas. We will consider these as we progress with the development work.

**Comment by rvanner** [comment]
Decals are being implemented in the latest version. There will also be bullet holes and other visual effects.

Closing this for now.

**Comment by fearlesswee** [comment]
Was going through my old github issues to see which ones should be updated/have been replied to, I think this should be re-opened as we currently only have bullet holes, which for the record don't actually use decal projection, but rather spawn a flat plane and roughly orient it to the surface.

I still strongly think having projectable decals for blood, level details (ex: graffiti on walls, mold/mildew, grime/dirt, etc.) should be considered.

**Comment by OldFlak** [comment]
The decals also need to be implemented properly - being able to project only to a single flat square plane is all but useless.
At present it is a half-baked implementation. 

Classic does it properly - MAX should also.

**Comment by LeeBamberTGC** [comment]
@fearlesswee @OldFlak Reopening this as a feature request for MAX. Classic did it a little smarter by detecting the extent of the polygon and only placing the flat quad within the borders, MAX just dumps the quad at the hit site.  Projection is a better way, and I believe Wicked does provide projection support out of the box.  Not sure on performance, but as this will be a sparingly used feature in the overall rendering system it may not impact on FPS at all (and worth checking out).  I also agree we need decals for things like blood, scorches, bullet holes and other needs, we only need to replace the decals that are pretending to be particles (for VR).

**Comment by OldFlak** [comment]
Thanks Lee:
There is a good example in this thread #3579

**Comment by mav3r1ck1981** [comment]
bumping this feature as it is becoming increasingly useful to multiple users including myself
